src/main/java/com/itproject/petshome/controller/VolunteerController.java

```diff
@@ -0,0 +1,31 @@
+package com.itproject.petshome.controller;
+
+import com.itproject.petshome.dto.VolunteerApplicationDTO;
+import com.itproject.petshome.exception.UserNotFoundException;
+import com.itproject.petshome.model.User;
+import com.itproject.petshome.service.SessionService;
+import com.itproject.petshome.service.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import lombok.AllArgsConstructor;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@SecurityRequirement(name = "bearerAuth")
+@AllArgsConstructor
+@RestController
+@Validated
+@RequestMapping("/api/v1/user")
+public class VolunteerController {
+    private UserService userService;
+    private SessionService sessionService;
+
+    @Operation(summary = "apply to volunteer")
+    @PostMapping("/applyToVolunteer")
+    public VolunteerApplicationDTO addVolunteerApplication()  throws UserNotFoundException {
```

Comment:
request body should be here(eg. reasons)

Reply:
I thought only an expression of interest is needed to apply to be a volunteer.

If volunteer application requires reasons and other information, then maybe VolunteerApplicationInput class is needed and also VolunteerApplication in model needs to be changed.